### ISSUE TITLE
[Enhancement] Can we please have option to have sidebar not show in a specific window if there is only one tab open?  (Perhaps exists but I can't find it!)

### ISSUE THREAD

**Issue by BensonBear** [piroor/treestyletab#3424]
I have one main firefox window with lots of tabs, but then I have some webapps (mail, editor, etc) that I want in a separate window, and when firefox starts I have to manually hide all the sidebars.   Would be nice if it did it for me.   No need for a sidebar if you intend to have only one tab open in the particular window.

I would not be surprised if this option is already in the long list of options, but I could not find it.

**Comment by SArpnt** [comment]
i think you want this
 
https://github.com/piroor/treestyletab/wiki/Code-snippets-for-custom-style-rules#auto-hide-sidebar-in-windows-with-only-one-tab-requires-another-extension-1768

**Comment by BensonBear** [comment]
Thanks a lot, that works great.

I first tried to avoid using the extra extension, since I know the title of the window which will be in the single tab, and only in the single tab.   But it did not seem to positively match its title, so perhaps this "prefix" is something special that has to be set, rather than literally being the start of the window title.  

But with this information I can probably figure it out (or just use it as is, although I am getting up there with number of extensions installed and would prefer to avoid adding more).

**Comment by piroor** [comment]
> I have one main firefox window with lots of tabs, but then I have some webapps (mail, editor, etc) that I want in a separate window, and when firefox starts I have to manually hide all the sidebars. Would be nice if it did it for me. No need for a sidebar if you intend to have only one tab open in the particular window.

Sadly it is impossible by TST because the initial visibility state of the sidebar on Firefox windows are not controllable via WebExtensions API. You need to use something hacks with userChrome.css or userChrome.js.

**Comment by BensonBear** [comment]
Thanks.  It seems maybe that it could be done using the same idea expressed above (which requires altering the window titles, and yes, one would still have to add some code to userChrome to effect the hiding), but it doesn't matter since that idea can be implemented separately with another extension and  it works fine.  Thanks! 
 
Maybe for anyone who comes across this and see's "wont fix", you could have another tag "external fix available" ("firefox-issue" suggests it's external, but doesn't hint that it can be solved externally without much difficulty).

**Comment by MurzNN** [comment]
To have individual windows for specific tabs without the sidebar - I use this extension https://github.com/ettoolong/PopupWindow - very convenient!

**Comment by irvinm** [comment]
@piroor this should likely be closed.  Please review.

**Comment by piroor** [comment]
I've researched again to consider possibility to implement the feature with WebExtensions way, and finally I realized that it is impossible. The APIs to open/close sidebar are not callable except handlers for the toolbar button or keyboard shortcuts.
https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/sidebarAction/open
https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/sidebarAction/close
The restriction is clearly designed behavior, so I think it won't be loosen. I hope to keep this issues tracker as the place to track unresolved problems possible resolved, so I close this for now.
